python/dgl/graphbolt/item_sampler.py

```diff
@@ -326,8 +329,108 @@ def __iter__(self) -> Iterator:
         data_pipe = data_pipe.collate(collate_fn=self._collate)
 
         # Map to minibatch.
-        data_pipe = data_pipe.map(
-            partial(self._minibatcher, names=self._item_set.names)
+        data_pipe = data_pipe.map(partial(self._minibatcher, names=self._names))
+
+        return iter(data_pipe)
+
+
+@functional_datapipe("distributed_sample_item")
+class DistributedItemSampler(ItemSampler):
+    """Distributed Item Sampler.
+
+    This is useful with torch's DDP training. It will create distributed item
+    subset of data which could be node IDs, node pairs with or without labels,
+    node pairs with negative sources/destinations, DGLGraphs and heterogeneous
+    counterparts. The items of the original item set will be sharded and form
+    an exclusive subset for each replica (process).
+
+    Note: This class `DistributedItemSampler` is not decorated with
+    `torchdata.datapipes.functional_datapipe` on purpose. This indicates it
+    does not support function-like call. But any iterable datapipes from
+    `torchdata` can be further appended.
+
+    Parameters
+    ----------
+    item_set : ItemSet or ItemSetDict
+        Data to be sampled.
+    batch_size : int
+        The size of each batch.
+    minibatcher : Optional[Callable]
+        A callable that takes in a list of items and returns a `MiniBatch`.
+    drop_last : bool
+        Option to drop the last batch if it's not full.
+    shuffle : bool
+        Option to shuffle before sample.
+    num_replicas: int
+        Number of processes participating in distributed training. By default,
+        `world_size` is retrieved from the current distributed group.
+    even_inputs : bool
```

Comment:
even_inputs is not very descriptive, what about drop_uneven_batches?
```suggestion
    drop_uneven_batches : bool
```